flashing: do not give bossac the offset parameter if not explicitly provided.

KubaFYI opened the pull request:
Doesn't feed bossac the offset parameter unless explicitly requested. This allows the runner to play nicely with earlier versions of bossac.

Fixes: #22171

Signed-off-by: Kuba Sanak <contact@kuba.fyi>

mbolivar commented:
For future reference, please force-push into your existing branches instead of closing a pull request and opening a new one. Closing PRs makes it harder for reviewers to track comments and their resolution. See the zephyr contribution guidelines for more details.

mbolivar commented on scripts/west_commands/runners/bossac.py:
```diff
@@ -56,8 +56,9 @@ def do_run(self, command, **kwargs):
-                     '-o', '%s' % self.offset,
                      '-b', self.cfg.bin_file]
+        if self.offset is not None:
+            cmd_flash = cmd_flash + ['-o', '%s' % self.offset]
```
Minor nit, an in-place update would be slightly preferable style-wise:

```suggestion
            cmd_flash += ['-o', '%s' % self.offset]
```

mbolivar reviewed (approved):
@galak up to you if this covers what you asked for in the other PR; it's fine with me

mbolivar commented:
@KubaFYI please take a look at the shippable error and make sure the tests are passing. See the README in the west_commands directory for info on running tests.

KubaFYI committed:
flashing: don't give bossac offset parameter unless explicitly provided

This fixes a problem which appeared after bossac version was downgraded
to 1.7 which no longer accepts the -o/--offset parameter. Now the offset
is fed to bossac executable only if it's explicitly provided and not by
default.

Signed-off-by: Kuba Sanak <contact@kuba.fyi>
